docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -460,6 +468,45 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best practice
+    :class: hint
+
+    Make the user experience of the operator as smooth as possible. The
+    application should be deployable without providing any further configuration
+    options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
+.. admonition:: Best practice
+    :class: hint
+
+    Just like Juju, a charm is an opinionated tool. Configure the application
+    with the best defaults, and only expose application configuration options
+    when necessary.
+
+.. admonition:: Best practice
+    :class: hint
+
+    Prefer lowercase alphanumeric names, separated with dashes if required. For
+    charms that have already standardised on underscores, it is not necessary to
+    change them, and it is better to be consistent within a charm then to have
+    some config names be dashed and some be underscored.
+
+.. admonition:: Best practice
+    :class: hint
+
+    The description is a string type. YAML supports two types of formats for
+    that: ``block scalar`` and ``flow scalar``. Prefer the block style
+    (using ``|``) to keep newlines. Using ``>`` will replace newlines with
+    spaces and make the result harder to read on Charmhub.
```

Comment:
I would cut this. It's more like a nice-to-know for a how-to guide than essential text in a reference, in my opinion.